internal/providers/mock.go

```diff
@@ -41,6 +41,19 @@ func (m *Mock) GetProviderSchema() GetProviderSchemaResponse {
 	if m.schema == nil {
 		// Cache the schema, it's not changing.
 		schema := m.Provider.GetProviderSchema()
+
+		// Override the provider schema with the constant mock provider schema.
+		// This is empty at the moment, check configs/mock_provider.go for the
+		// actual schema.
```

Comment:
I'd suggest a note in this comment saying that this write is safe because `schema` is a `providers.GetProviderSchemaResponse` object passed by value, and so no other codepaths can possibly be observing that object.

This activated my "spidey sense" for situations where we've accidentally modified shared objects before and caused me to go check to make sure this was safe, which it does seem to be but I'd prefer to have that assumption documented here to aid future maintainers so they can know to adjust this if, for example, we add some additional schema-patching here in future that _would_ cause a shared object to be modified.

Reply:
Done!